src/serac/physics/integrators/traction_integrator.cpp

```diff
@@ -0,0 +1,226 @@
+
+void PressureIntegrator::AssembleFaceVector(const mfem::FiniteElement&        element_1, const mfem::FiniteElement&,
+                                            mfem::FaceElementTransformations& parent_to_reference_face_transformation,
+                                            const mfem::Vector&               input_state_vector,
+                                            mfem::Vector&                     output_residual_vector)
+{
+  // Get the dimension and number of degrees of freedom from the current element
+  int dim = element_1.GetDim();
+  int dof = element_1.GetDof();
+
+  // Ensure the working data structures are sized appropriately
+  shape_.SetSize(dof);
+  output_residual_vector.SetSize(dim * dof);
+  dN_dxi_.SetSize(dof, dim);
+  dN_dX_.SetSize(dof, dim);
+  dxi_dX_.SetSize(dim);
+  F_.SetSize(dim);
+  Finv_.SetSize(dim);
+  reference_normal_.SetSize(dim);
+  current_normal_.SetSize(dim);
+
+  // Reshape the input state as a matrix (dof x dim)
+  input_state_matrix_.UseExternalData(input_state_vector.GetData(), dof, dim);
+
+  // Calculate an appropriate integration rule for the element order
+  int                          intorder = 2 * element_1.GetOrder() + 3;
+  const mfem::IntegrationRule& ir = mfem::IntRules.Get(parent_to_reference_face_transformation.FaceGeom, intorder);
+
+  // Initialize the residual output
+  output_residual_vector = 0.0;
+
+  for (int i = 0; i < ir.GetNPoints(); i++) {
+    // Set the current integration point
+    const mfem::IntegrationPoint& ip = ir.IntPoint(i);
+    mfem::IntegrationPoint        eip;
+    parent_to_reference_face_transformation.Loc1.Transform(ip, eip);
+    parent_to_reference_face_transformation.Face->SetIntPoint(&ip);
+
+    // Calculate the normal vector in the reference configuration
+    CalcOrtho(parent_to_reference_face_transformation.Face->Jacobian(), reference_normal_);
+    double norm = reference_normal_.Norml2();
+    reference_normal_ /= norm;
+
+    if (compute_on_reference_) {
+      current_normal_ = reference_normal_;
+    } else {
+      // Calculate the deformation gradient and its inverse
+      parent_to_reference_face_transformation.Elem1->SetIntPoint(&eip);
+      CalcInverse(parent_to_reference_face_transformation.Elem1->Jacobian(), dxi_dX_);
+      element_1.CalcDShape(eip, dN_dxi_);
+      Mult(dN_dxi_, dxi_dX_, dN_dX_);
+      MultAtB(input_state_matrix_, dN_dX_, F_);
+
+      for (int d = 0; d < dim; d++) {
+        F_(d, d) += 1.0;
+      }
+
+      CalcInverse(F_, Finv_);
+
+      Finv_.MultTranspose(reference_normal_, current_normal_);
+      current_normal_ *= F_.Det();
+    }
+
+    element_1.CalcShape(eip, shape_);
+    current_normal_ *= pressure_.Eval(*parent_to_reference_face_transformation.Face, ip);
+
+    for (int j = 0; j < dof; j++) {
+      for (int k = 0; k < dim; k++) {
+        output_residual_vector(dof * k + j) +=
+            ip.weight * parent_to_reference_face_transformation.Face->Weight() * current_normal_(k) * shape_(j);
```

Comment:
Good to have a comment "current_normal has area transformation and includes the applied pressure, and is therefore not normalized"